src/Driver/AbstractMySQLDriver.php

```diff
@@ -126,6 +127,8 @@ public function getDatabasePlatform()
      */
     public function getSchemaManager(Connection $conn, AbstractPlatform $platform)
     {
+        assert($platform instanceof MySQLPlatform);
```

Comment:
Would it make sense to add the platform class as a `@template` to the `Driver` interface instead, like you did for `AbstractSchemaManager`?

Reply:
It would. I tried that before wrapping up the patch but it looks like it's a rabbit hole (this is the current state: https://github.com/morozov/dbal/commit/8990107aae96b91d8a5b5362aa936c5347db245c).

A certain complexity arises when we parameterize the types like `Connection`, `Driver`, etc. that are used in connection parameters.

We can tackle this issue separately.

Reply:
Fair enough. 👍🏻